packages/front-end/components/Features/requestReviewModal.tsx

```diff
@@ -0,0 +1,371 @@
+            b: JSON.stringify(result.rules[env.id], null, 2),
+          });
+        }
+      });
+    }
+
+    return diffs;
+  }, [
+    environments,
+    feature.defaultValue,
+    feature.environmentSettings,
+    mergeResult,
+  ]);
+
+  if (!revision || !mergeResult) return null;
+
+  const hasPermission = permissions.check(
+    "publishFeatures",
+    feature.project,
+    getAffectedRevisionEnvs(feature, revision, environments)
+  );
+
+  const hasChanges = mergeResultHasChanges(mergeResult);
+  let ctaCopy = "Request Review";
+  if (approved) {
+    ctaCopy = "Publish";
+  } else if (canReview) {
+    ctaCopy = "Next";
+  }
+  const renderRequestAndViewModal = () => {
+    return (
+      <Modal
+        open={true}
+        header={"Review Draft Changes"}
+        cta={ctaCopy}
+        close={close}
+        autoCloseOnSubmit={canReview ? false : true}
+        closeCta="Cancel"
+        size="max"
+        submit={
+          !isPendingReview || canReview || approved ? submitButton : undefined
+        }
+        secondaryCTA={
+          isPendingReview && !canReview ? (
+            <LegacyButton
+              color="danger"
+              onClick={async () => {
+                try {
+                  await apiCall(
+                    `/feature/${feature.id}/${revision.version}/discard`,
+                    {
+                      method: "POST",
+                    }
+                  );
+                } catch (e) {
+                  await mutate();
+                  throw e;
+                }
+                await mutate();
+                onDiscard && onDiscard();
+                close();
+              }}
+            >
+              Discard
+            </LegacyButton>
+          ) : undefined
+        }
+      >
+        {mergeResult.conflicts.length > 0 && (
+          <div className="alert alert-danger">
+            <strong>Conflicts Detected</strong>. Please fix conflicts before
+            publishing this draft.
+          </div>
+        )}
+
+        {!hasChanges && !mergeResult.conflicts.length && (
+          <div className="alert alert-info">
+            There are no changes to publish. Either discard the draft or add
+            changes first before publishing.
+          </div>
+        )}
+
+        {mergeResult.success && hasChanges && (
+          <div>
+            <Callout.Root color={isPendingReview ? "amber" : "gray"}>
+              <Callout.Text>
+                Publishing to the prod environment requires approval.
+              </Callout.Text>
+            </Callout.Root>
+            {isAdmin && (
+              <Text as="label" size="2" className="mt-2">
+                <Flex gap="2">
+                  <Checkbox
+                    onCheckedChange={(checkedState) => {
+                      checkedState === "indeterminate"
+                        ? setAdminPublish(false)
+                        : setAdminPublish(checkedState);
+                    }}
+                  />
```

Comment:
This has some weird behavior.  When I first check this, it flashes for a second and then goes back to unchecked, but the  "publish" button shows up